### PR TITLE
Tracking tx-id per network, with accessors

### PR DESCRIPTION
**Pull request by kingcocomango**
Part of three PRs for https://github.com/trufflesuite/truffle/issues/472

### CODE REVIEW

File: contract.js [hunk 1]
```diff
@@ -745,6 +745,23 @@ var contract = (function(module) {
...
+
+        return transactionHash;
+      },
+      set: function(val) {
+        if(val === null) {
+          throw new Error(`Could not set \`${val}\` as the transaction ID for ${this.contractName}`);
+        }
+        this.network.transactionId = val;
+      }
+    },
```
Comment (gnidan) [inline contract.js]: Same as the truffle-deployer PR, mind updating the naming here to just `transactionId` instead of `transactionHash`?

Comment (owocki) [reply contract.js]: @kingcocomango ^^

**Comment by cgewecke** [comment]
@kingcocomango I don't want to alarm you but in the end we talked it over and decided that everything should be named `transactionHash`, because get that property from `web3` and it makes sense to be consistent about naming across the assets, even though `ID` is nicer in some ways. Have updated the three PRs to that end and am pinging @gnidan right now about resolving the bounty for you.

**Comment by kingcocomango** [comment]
No problem, I agree that's why I'd initially went with that name :)

**Comment by cgewecke** [comment]
@kingcocomango :) Thanks for your forbearance with us, really appreciate you doing this. Nick has told GitCoin this is done, please just let us know if that side of it doesn't resolve shortly.

**Pull request merged**